src/tests/cmocka/test_negcache_2.c

```diff
@@ -33,6 +33,10 @@
 #include "responder/common/negcache.h"
 #include "responder/common/responder.h"
 
+#ifndef __GLIBC__
+#include "tests/musl-passwd.c.inc"
```

Comment:
I really wouldn't like to pull this into the codebase.

Frankly, I don't see much point to bother with special tests for locally found users (and,btw, test will likely fail if there are less than 2 non-root users in /etc/passwd)
https://github.com/SSSD/sssd/blob/38a6a4a79226a784eb2233203f74cf2040e67856/src/tests/cmocka/test_negcache_2.c#L335

In cf650328632ae25ad7c8947918988500081b7822 I have already removed some files_provider / local users related test code, maybe more can be removed so there is no need to parse /etc/passwd & group

If you neither want to dive into this nor wait until somebody do so, then maybe you could better cut this test (or some its cases) off at compile time on FreeBSD?